### PR TITLE
SDK-432 [C# SDK] refactor README

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic (if any):
- [SDK-432](https://incountry.atlassian.net/browse/SDK-432)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1104,9 +1115,9 @@ public void Test() {
 ```
 
 ## Custom encryption support
-SDK supports the ability to provide custom encryption/decryption methods if you decide to use your own algorithm instead of the default one.
+The C# SDK supports a capability to provide custom encryption/decryption methods if you decide to use your own algorithm instead of the default one.
```
Comment (RommeroInCountry) [inline README.md:1116]: The C# SDK allows to provide custom ...

File: README.md [hunk 2]
```diff
@@ -1184,17 +1192,16 @@ secrets_data = {
 
 ---
 
-Parameter `name` of `AbstractCipher` constructor is used to differ one custom encryption from another and from the default encryption as well. This way SDK will be able to successfully decrypt any old data if encryption changes with time.
+The name parameter of the `AbstractCipher` constructor is used to differentiate custom encryptions from each other and from the default encryption as well. This way the SDK is able to successfully decrypt any old data if encryption changes over time.
```
Comment (RommeroInCountry) [inline README.md]: The `name` parameter

Comment (RommeroInCountry) [reply README.md]: to differentiate custom encryption implementations from each other

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

File: InCountrySDK/IStorage.cs [hunk 3]
```diff
@@ -13,161 +13,162 @@ namespace InCountrySDK
...
+        /// <param name="filter">The object containing the find filters and search options</param>
+        /// <exception cref="Exception.StorageClientException">When validation of parameters fails</exception>
+        /// <exception cref="Exception.StorageServerException">When server connection fails or upon a server response error</exception>
+        /// <exception cref="Exception.StorageCryptoException">When decryption fails</exception>
+        /// <returns>Found record or null</returns>
         Task<Record> FindOneAsync(string country, FindFilter filter = null);
 
         /// <summary>
-        /// Make asynchronously batched key-rotation-migration of records
+        /// Performs an asynchronous batch key rotation migration of records
```
[the remark this answers is not in the document]
Comment (VladimirLevitskiy) [reply InCountrySDK/IStorage.cs:91]: latest

File: README.md (hunk 1 above)
Comment (VladimirLevitskiy) [reply README.md:1116]: allows providing

**Review by RommeroInCountry — APPROVED** [review]

**Review by RommeroInCountry — APPROVED** [review]

**Review by RommeroInCountry — APPROVED** [review]

**Pull request merged**